tests/integration/test_reset_subdomains.cairo

```diff
@@ -62,3 +79,50 @@ func test_reset_subdomains{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: Ha
 
     return ();
 }
+
+@external
+func test_reclaim_subdomain{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: HashBuiltin*}() {
+    alloc_locals;
+    local starknet_id_contract;
+    local naming_contract;
+    %{
+        ids.starknet_id_contract = context.starknet_id_contract
+        ids.naming_contract = context.naming_contract
+        stop_prank_callable = start_prank(456)
+        stop_mock = mock_call(123, "transferFrom", [1])
+        warp(1, context.naming_contract)
+    %}
+
+    let token_id = 1;
+    StarknetId.mint(starknet_id_contract, token_id);
+
+    let token_id2 = 2;
+    StarknetId.mint(starknet_id_contract, token_id2);
+
+    // should mint a domain and create a subdomain
+    Naming.buy(naming_contract, token_id, 'alpha', 365, 0, 456);
+
+    // should mint a subdomain
+    Naming.transfer_domain(naming_contract, 2, new ('bravo', 'alpha'), token_id2);
+    let (token_id) = Naming.domain_to_token_id(naming_contract, 2, new ('bravo', 'alpha'));
+    assert token_id = token_id2;
+
+    // should reset the subdomain owner
+    Naming.reset_subdomains(naming_contract, 1, new ('alpha'));
+    let (token_id) = Naming.domain_to_token_id(naming_contract, 2, new ('bravo', 'alpha'));
+    assert token_id = 0;
+
+    let token_id3 = 3;
+    StarknetId.mint(starknet_id_contract, token_id3);
+    // should mint the subdomain again
```

Comment:
You have to place this before the declaration of token_id3 in order for it to be clearer ;)